addons/mail/models/mail_channel.py

```diff
@@ -239,6 +239,10 @@ def _unlink_except_all_employee_channel(self):
             raise UserError(_('You cannot delete those groups, as the Whole Company group is required by other modules.'))
 
     def write(self, vals):
+        if 'channel_type' in vals:
+            for channel in self:
+                if channel.channel_type:
+                    raise UserError(_('Cannot change channel type because \'%s\' is already been set to \'%s\'.') % (channel.name, channel.channel_type))
```

Comment:
Pending rebase on the other PR that was created following this issue, but there should be no condition on channel_type, it should always be set.